src/mtcp/mtcp_restart.c

```diff
@@ -495,13 +496,92 @@ mysetauxval(char **evp, unsigned long int type, unsigned long int val)
   return -1;
 }
 
-#define PAGESIZE 4096 /* We hardwire this, since we can't make libc calls. */
-// We choose this address based on our observation on CORI that it won't overlap
-// with any other memory segments. We might change this address in the future,
-// for now, it doesn't matters.
-#define vvarStartTmp ((0x40000) - 5*PAGESIZE)
-#define vdsoStartTmp ((0x40000) - 2*PAGESIZE)
+/* We hardwire these, since we can't make libc calls. */
+#define PAGESIZE 4096
+#define ROUNDADDRUP(addr, size) ((addr + size - 1) & ~(size - 1))
+// We use global variables here, since both remap_vdso_and_vvar_regions and
+// unmap_memory_areas_and_restore_vdso use these variables. They're used to
+// track the temporary regions the vvar and vdso segments are remapped to, so
+// they are not unmapped later.
+uint64_t vvarStartTmp = 0;
+uint64_t vdsoStartTmp = 0;
```

Comment:
Add `static` if they will only be used in this file.

Reply:
adding `static` strangely causes the `vdso/vvar order was different during ckpt` error.